ui/ducks/confirm/confirm.ts

```diff
@@ -15,19 +22,35 @@ const initState = {
 };
 
 export default function confirmReducer(
-  state = initState,
-  action: ActionType = { type: '' },
+  // eslint-disable-next-line @typescript-eslint/default-param-last
+  state: typeof initState = initState,
+  action: ActionType,
 ) {
   switch (action.type) {
-    case UPDATE_CURRENT_CONFIRMATION:
+    case 'metamask/confirm/UPDATE_CONFIRM': {
+      return {
+        ...state,
+        ...action.payload,
+      };
+    }
+    case 'metamask/confirm/UPDATE_CURRENT_CONFIRMATION': {
       return {
+        ...state,
         currentConfirmation: action.currentConfirmation,
       };
+    }
     default:
       return state;
   }
 }
 
+export function updateConfirm(payload: Record<string, unknown> | undefined) {
+  return {
+    type: UPDATE_CONFIRM,
+    payload: payload || {},
+  };
+}
+
```

Comment:
May be `payload || initState`

Reply:
lgtm! thank you 
updated b78eb1e3b030b5a22cbd2acd279fd9b114c64e15